xapian-letor/api/letor.cc

```diff
@@ -64,8 +64,8 @@ Letor::set_query(const Xapian::Query & query) {
 }
 
 std::vector<Xapian::docid>
-Letor::letor_rank(const Xapian::MSet & mset, Xapian::FeatureList & flist, const char* model_filename) {
-    return internal->letor_rank(mset, flist, model_filename);
+Letor::letor_rank(const Xapian::MSet & mset, const char* model_filename, Xapian::FeatureList & flist) {
```

Comment:
[This is getting quite future speculative, so we don't have to worry about it yet. However:]

Ideally I think we'd just let people call it on `Ranker` directly. The only problem is for the default ranker as we currently do things. However I'm wondering if we should actually be storing the model as database metadata instead of as a file (so it'll get replicated, for instance). Then we could have a default metadata key name for each ranker type, which would take care of the default, or you could specify a different key (in the ranker constructor, probably) if you wanted to have different models trained with the same algorithm. So:

```
Ranker ranker = ListNET(); // Uses default, loads model from key letor.models.listnet or similar
Ranker ranker2 = ListNET("my.listnet.model"); // Loads a different model
```